project/data_analysis/python/so/so_prepare_sim_data.py

```diff
@@ -0,0 +1,347 @@
+    "Planck_143xPlanck_143": 33.0,
+    "Planck_217xPlanck_217": 46.80,
+    "Planck_353xPlanck_353": 153.6
+}
+
+sigma_pol = {
+    "Planck_100xPlanck_100": 117.6,
+    "Planck_143xPlanck_143": 70.2,
+    "Planck_217xPlanck_217": 105.0,
+    "Planck_353xPlanck_353": 438.6
+}
+
+n_ell_t_Planck_dict = {}
+n_ell_pol_Planck_dict = {}
+for f_pair_Planck in f_pairs_Planck:
+    sigma_t_rad = np.deg2rad(sigma_t[f_pair_Planck]) / 60
+    sigma_pol_rad = np.deg2rad(sigma_pol[f_pair_Planck]) / 60
+
+    n_ell_t_Planck_dict[f_pair_Planck] = np.full_like(ell, sigma_t_rad ** 2)
+    n_ell_pol_Planck_dict[f_pair_Planck] = np.full_like(ell, sigma_pol_rad ** 2)
+
+# Fill the dict with non-zero noise power
+for f_pair_Planck in f_pairs_Planck:
+    if f_pair_Planck in spec_name_list:
+        n_ell_t[f_pair_Planck] = n_ell_t_Planck_dict[f_pair_Planck]
+        n_ell_pol[f_pair_Planck] = n_ell_pol_Planck_dict[f_pair_Planck]
+
+# Now let's write the n_ell_t and n_ell_pol dictionnary to disk
+# Note that we are creating a lot of small files, we could use another data format
+if d["type"] == "Dl":
+    fac = ell * (ell + 1) / (2 * np.pi)
+else:
+    fac = 1.
+
+nlth_dict = {}
+for sv in surveys:
+    for id1, ar1 in enumerate(d[f"arrays_{sv}"]):
+        for id2, ar2 in enumerate(d[f"arrays_{sv}"]):
+            if id1 > id2: continue
+            spec_name = f"{sv}_{ar1}x{sv}_{ar2}"
+            mean_noise_t = n_ell_t[spec_name] * fac
+            mean_noise_p = n_ell_pol[spec_name] * fac
+            mean_noise = {"TT": mean_noise_t, "EE": mean_noise_p, "BB": mean_noise_p}
+            for spec in spectra:
+                if spec in ["TT", "EE", "BB"]: continue
+                mean_noise[spec] = np.zeros_like(ell)
+
+
+            # apply a regularization to the noise power spectrum at very low ell
+            l_cut_noise = d["l_cut_noise_LAT"]
+            for spec in spectra:
+                mean_noise[spec] = np.where(ell <= l_cut_noise, 0., mean_noise[spec])
+            nlth_dict[spec_name] = mean_noise
+
+            so_spectra.write_ps(f"{noise_model_dir}/mean_{ar1}x{ar2}_{sv}_noise.dat", ell, mean_noise, type, spectra = spectra)
+            so_spectra.write_ps(f"{noise_model_dir}/mean_{ar2}x{ar1}_{sv}_noise.dat", ell, mean_noise, type, spectra = spectra)
+
+# Finally let's generate the beam harmonic transform for Planck and SO LAT
+# For Planck we  we will use information from the Table 4 of
+# https://arxiv.org/pdf/1807.06205.pdf
+# For SO we use info from Table 1 of
+# https://arxiv.org/pdf/1808.07445.pdf
+beam_plot_dir = f"{plot_dir}/beams"
+pspy_utils.create_directory(beam_plot_dir)
+
+beam_fwhm = {
+    "LAT_27": 7.4,
+    "LAT_39": 5.1,
+    "LAT_93": 2.2,
+    "LAT_145": 1.4,
+    "LAT_225": 1.0,
+    "LAT_280": 0.9,
+    "Planck_100": 9.68,
+    "Planck_143": 7.30,
+    "Planck_217": 5.02,
+    "Planck_353": 4.94
+}
+
+l = np.arange(ell_max + 1000)
+bl = {f"{sv}_{ar}": np.ones(ell_max + 1000) for sv,ar in zip(sv_list, ar_list)}
```

Comment:
since you are going to use `zip(sv_list, ar_list)` a lot in the forthcoming `for` loop maybe you can define a variable with directly the results something like
```
sv_ar_list = [f"{sv}_{ar}" for sv, ar in zip(sv_list, ar_list)]
```
and then the `for` loops will be
```
for sv_ar in sv_ar_list:
```